src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -891,19 +893,122 @@ public void testLookUpFieldList() throws IOException, CommandException
+        getDriver().switchTo().alert().dismiss();
+
+        // make sure we are still here
+        assertEquals(currentUrl, getDriver().getCurrentUrl());
+
+        // now navigate away
+        getDriver().navigate().to(homeUrl);
+        // dismiss the alert
+        shortWait().until(ExpectedConditions.alertIsPresent());
+        getDriver().switchTo().alert().dismiss();
+        // ensure current location
+        assertEquals(currentUrl, getDriver().getCurrentUrl());
+
+        domainDesignerPage = new DomainDesignerPage(getDriver()); // we've been refreshing; things are possibly stale
+        domainDesignerPage.clickCancel().saveChanges();
+    }
+
+    @Test
+    public void verifyWarningForMarkingExistingSampleSetFieldRequired() throws Exception
+    {
+        String sampleSetName = "hasRowsWithBlankValuesWarnSampleSet";
+
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", sampleSetName);
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("color", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("manufacturer", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("volume", FieldDefinition.ColumnType.Double)));
+        DomainResponse createResponse = dgen.createDomain(createDefaultConnection(true), "SampleSet");
+
+        dgen.addCustomRow(Map.of("name", "agar", "color", "green", "manufacturer", "glaxo", "volume", 2.34));
+        dgen.addCustomRow(Map.of("name", "stuff", "color", "clear", "manufacturer", "glaxo", "volume", 2.34));
+        dgen.addCustomRow(Map.of("name", "icecream", "color", "blue", "manufacturer", "glaxo", "volume", 2.34));
+        dgen.addCustomRow(Map.of("name", "pbs", "color", "yellow", "manufacturer", "glaxo", "volume", 2.34));
+        dgen.addCustomRow(Map.of("name", "slurm", "color", "orange",                                  "volume", 2.34));  //<-- no value for manufacturer
+        dgen.insertRows(createDefaultConnection(true), dgen.getRows());
+
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "exp.materials", sampleSetName);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldProperties(sampleSetName);
 
+        // now attempt to mark 'manufacturer' field 'required'
+        domainFormPanel.getField("manufacturer")
+                .setRequiredField(true);
+        domainDesignerPage.clickSave();     // expect error warning here; this should warn the user
+        String expectedWarning = domainDesignerPage.waitForAnyAlert();
+        assertTrue(expectedWarning.contains("cannot be required when it contains rows with blank values."));
```

Comment:
Do we want a test case that verifies you **can** mark an existing field as required if there aren't any rows with blank values?

Reply:
That's not a bad idea.  I'll add it